src/serverenvironment.cpp

```diff
@@ -2122,6 +2009,69 @@ void ServerEnvironment::deactivateFarObjects(bool _force_delete)
 	}
 }
 
+void ServerEnvironment::deleteStaticFromBlock(
+		ServerActiveObject* obj, u16 id, u32 mod_reason, bool no_emerge)
+{
+	if (!obj->m_static_exists)
+		return;
+
+	MapBlock *block;
+	if (no_emerge)
+		block = m_map->getBlockNoCreateNoEx(obj->m_static_block);
+	else
+		block = m_map->emergeBlock(obj->m_static_block, false);
+	if (!block) {
+		if (!no_emerge)
+			errorstream << "ServerEnv: Failed to emerge block " << PP(obj->m_static_block)
+					<< " when deleting static data of object from it. id=" << id << std::endl;
+		return;
+	}
+
+	block->m_static_objects.remove(id);
+	if (mod_reason != MOD_REASON_UNKNOWN) // Do not mark as modified if requested
+		block->raiseModified(MOD_STATE_WRITE_NEEDED, mod_reason);
+
+	obj->m_static_exists = false;
+}
+
+bool ServerEnvironment::saveStaticToBlock(
+		v3s16 blockpos, u16 store_id,
+		ServerActiveObject* obj, const StaticObject &s_obj,
+		u32 mod_reason)
+{
+	MapBlock *block = NULL;
```

Comment:
nullptr please

Reply:
`emergeBlock` will return `NULL` not `nullptr` in case of failure
both shouldn't be mixed, right?

Reply:
as you check if (!block) NULL or nullptr is not a problem, it can be a problem if you check == NULL